[BUG] `import/no-unused-modules` does not work with `import { name as otherName }`

When using `import { ... as ... }` statements, `import/no-unused-modules` does not seem to properly work.

In the following example, `test` export is used, but it's still reported as not used:

```
$ eslint .
/home/ehmicky/no-unused-modules-bug/src/file.js
  1:1  error  exported declaration 'test' not used within other modules  import/no-unused-modules

✖ 1 problem (1 error, 0 warnings)
```

`src/main.js`:

```js
import { test as testTwo } from './file.js'
```

`src/file.js`:

```js
export const test = true
```

`.eslintrc.json`:

```json
{
  "parserOptions": {
    "ecmaVersion": 2019,
    "sourceType": "module"
  },
  "plugins": ["import"],
  "rules": {
    "import/no-unused-modules": [
      2,
      {
        "unusedExports": true,
        "ignoreExports": ["src/main.js"]
      }
    ]
  }
}
```

`package.json`:

```json
{
  "name": "no-unused-modules-bug",
  "version": "1.0.0",
  "dependencies": {},
  "devDependencies": {
    "eslint": "^5.16.0",
    "eslint-plugin-import": "^2.17.2"
  }
}
```

Note: removing the `ignoreExports: ["src/main.js"]` does not fix this issue.

--- a/src/rules/no-unused-modules.js
+++ b/src/rules/no-unused-modules.js
@@ -503,7 +503,7 @@ module.exports = {
                 specifier.type === IMPORT_NAMESPACE_SPECIFIER) {
               return
             }
-            newImports.set(specifier.local.name, resolvedPath)
+            newImports.set(specifier.imported.name, resolvedPath)
           })
         }
       })

--- a/tests/src/rules/no-unused-modules.js
+++ b/tests/src/rules/no-unused-modules.js
@@ -254,6 +254,34 @@ ruleTester.run('no-unused-modules', rule, {
   ],
 })
 
+// add renamed named import for file with named export
+ruleTester.run('no-unused-modules', rule, {
+  valid: [
+    test({ options: unusedExportsOptions,
+           code: `import { g as g1 } from '${testFilePath('./no-unused-modules/file-g.js')}'; import eslint from 'eslint'`,
+           filename: testFilePath('./no-unused-modules/file-0.js')}),
+    test({ options: unusedExportsOptions,
+            code: 'export const g = 2',
+            filename: testFilePath('./no-unused-modules/file-g.js')}),
+    ],
+  invalid: [],
+})
+
+// add different renamed named import for file with named export
+ruleTester.run('no-unused-modules', rule, {
+  valid: [
+    test({ options: unusedExportsOptions,
+           code: `import { g1 as g } from '${testFilePath('./no-unused-modules/file-g.js')}'`,
+           filename: testFilePath('./no-unused-modules/file-0.js')}),
+  ],
+  invalid: [
+    test({ options: unusedExportsOptions,
+           code: 'export const g = 2',
+           filename: testFilePath('./no-unused-modules/file-g.js'),
+           errors: [error(`exported declaration 'g' not used within other modules`)]}),
+  ],
+})
+
 // remove default import for file with default export
 ruleTester.run('no-unused-modules', rule, {
   valid: [
@@ -556,7 +584,7 @@ describe('do not report missing export for ignored file', () => {
       test({ options: [{
                src: [testFilePath('./no-unused-modules/**/*.js')],
                ignoreExports: [testFilePath('./no-unused-modules/*ignored*.js')],
-               missingExports: true
+               missingExports: true,
               }],
              code: 'export const test = true',
              filename: testFilePath('./no-unused-modules/file-ignored-a.js')}),

